pandas/core/index.py

```diff
@@ -2543,11 +2550,12 @@ def insert(self, loc, item):
         -------
         new_index : Index
         """
-        _self = np.asarray(self)
-        item_idx = Index([item], dtype=self.dtype).values
-        idx = np.concatenate(
-            (_self[:loc], item_idx, _self[loc:]))
-        return Index(idx, name=self.name)
+        indexes=[self[:loc],
+                 Index([item]),
+                 self[loc:]]
+
+        return indexes[0].append(indexes[1]).append(indexes[2])
```

Comment:
But that leaves us with the same problem for PeriodIndex. This is the broader "`PeriodIndex` is not like the others I mentioned https://github.com/pydata/pandas/pull/10620#issuecomment-124182475

Or am I missing something obvious?

```python
In [171]:
pd.Index(np.asarray(period_index), **period_index._get_attributes_dict())
Out[171]:
Int64Index([11740, 11741, 11742, 11743, 11744, 11745, 11746, 11747, 11748,
            11749, 11750, 11751, 11752, 11753, 11754, 11755, 11756, 11757,
            11758, 11759, 11760, 11761],
           dtype='int64')
```